src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1609,6 +1698,150 @@ RebalanceTableShards(RebalanceOptions *options, Oid shardReplicationModeOid)
 }
 
 
+static bool
+HasConcurrentRebalance(RebalanceOptions *options)
+{
+	if (HasNonTerminalJobOfType("rebalance", NULL))
+	{
+		return true;
+	}
+
+	return false;
+}
+
+
+static void
+ErrorOnConcurrentRebalance(RebalanceOptions *options)
+{
+	Oid relationId = InvalidOid;
+	foreach_oid(relationId, options->relationIdList)
```

Comment:
do we need to sort this list? seems kind of deadlocky otherwise

Reply:
We didn't sort this in the old rebalancer, but wouldn't hurt I guess to sort before using for locks.

Reply:
Looked into this, the sorting would be very unintuitive, as it is not locking the relation, but instead the colocation id of the relation.

I wonder if it would be worth to refactor this code to _not_ pass random relations from colocation id's around, but instead actually pass colocation id's around, probably sorted, and work on them. Although that is quite some work.

Reply:
discussed offline, it's an existing issue in the rebalancer that we will not fix for now because the relation IDs are arbitrary IDs from the co-location groups